### PR TITLE
variable: add system variable 'tidb_resource_control_strict_mode'

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: ref #53440

Problem Summary:
Add new global variable `tidb_resource_control_strict_mode`:
- If set to 'ON', use need to have `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to execute `SET RESOURCE GROUP` or use hint `RESOURCE_GROUP()` to change current resource group.
- If set to 'OFF', the privilege is bypassed, all user can execute `SET RESOURCE GROUP` or use hint `RESOURCE_GROUP()`  as before.

For new cluster, the default value is 'ON', but for cluster upgraded from old version, the default value is 'OFF'.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add new global variable `tidb_resource_control_strict_mode` to config whether privilege check should be apply to operation that change current session/statement's resource group.
```

### CODE REVIEW

**Comment by glorv** [comment]
@nolouch  @tiancaiamao  PTAL, thanks~

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -2779,6 +2779,15 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(ctx context.Context, vars *SessionVars) (string, error) {
 		return BoolToOnOff(EnableResourceControl.Load()), nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBResourceControlStrictMode, Value: BoolToOnOff(DefTiDBResourceControlStrictMode), Type: TypeBool, SetGlobal: func(ctx context.Context, vars *SessionVars, s string) error {
+		if TiDBOptOn(s) != EnableResourceControl.Load() {
```
Comment (nolouch) [inline pkg/sessionctx/variable/sysvar.go:2783]:
```suggestion
		if TiDBOptOn(s) != EnableResourceControlStrictMode.Load() {
```
should be `EnableResourceControlStrictMode`?

Comment (glorv) [reply pkg/sessionctx/variable/sysvar.go:2783]: fixed. It's weird that unit test does not find this bug.

**Review by nolouch — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**